arangod/Aql/ExecutionBlockImpl.tpp

```diff
@@ -2560,6 +2559,8 @@ void ExecutionBlockImpl<Executor>::PrefetchTask::wakeupWaiter() noexcept {
   // need to temporarily lock the mutex to enforce serialization with the
   // waiting thread
   _lock.lock();
+  // (3) - this release-store synchronizes with the acquire-load (1, 2)
+  updateStatus(Status::Finished, std::memory_order_release);
```

Comment:
We need to also set the `abandoned` flag to ensure that the task gets rescheduled.
Since we are currently the owner of the task, no one else can modify it, so we can simply use `store` here:
```suggestion
  _state.store({Status::Finished, true}, std::memory_order_release);
```